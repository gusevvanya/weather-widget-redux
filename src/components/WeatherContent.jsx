import React from 'react';
import PropTypes from 'prop-types';

const WeatherContent = ({ currentCity, weather }) => {
  if (currentCity.length === 0) return null;

  return (
    <div className="weather">
      {currentCity}
      <br />
      {weather} &#176;F
    </div>
  );
};

WeatherContent.propTypes = {
  currentCity: PropTypes.string.isRequired,
  weather: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
};

export default WeatherContent;
