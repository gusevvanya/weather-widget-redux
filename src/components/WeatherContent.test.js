import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import WeatherContent from './WeatherContent';

Enzyme.configure({ adapter: new Adapter() });

describe('WeatherContent', () => {
  let props;
  let WeatherContentMount;
  let WeatherContentShallow;

  beforeAll(() => {
    props = {
      currentCity: 'London',
      weather: '23.43',
    };
  });

  beforeEach(() => {
    WeatherContentMount = Enzyme.mount(<WeatherContent {...props} />);
    WeatherContentShallow = Enzyme.shallow(<WeatherContent {...props} />);
  });

  it('should render', () => {
    expect(WeatherContentShallow.debug()).toMatchSnapshot();
  });

  it('render weather if the param "currentCity" recived', () => {
    expect(WeatherContentMount.find('.weather').length).toBe(1);
    WeatherContentMount.setProps({ currentCity: '' });
    expect(WeatherContentMount.find('.weather').length).toBe(0);
  });
});
