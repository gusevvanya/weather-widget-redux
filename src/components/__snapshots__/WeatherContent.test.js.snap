// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherContent should render 1`] = `
"<div className=\\"weather\\">
  London
  <br />
  23.43
   Â°F
</div>"
`;
